### PR TITLE
Feature/animation 16 18

### PR DESCRIPTION
**Pull request by DmitrySuprun**
https://user-images.githubusercontent.com/95674666/215446506-1efb547d-5f82-42d4-bcd7-04254410e901.mov

https://user-images.githubusercontent.com/95674666/215384173-b2cbf083-2630-4d4b-ac42-dd6057d361d1.mov

https://user-images.githubusercontent.com/95674666/215384186-7e101b84-f53b-49ab-8547-9a6a4062d29b.mov

https://user-images.githubusercontent.com/95674666/215384189-6ba792d1-70b7-4447-b115-77f7df248081.mp4

https://user-images.githubusercontent.com/95674666/215384191-9dbab572-a4c0-4e43-b65b-312f99895f59.mp4

### CODE REVIEW

File: RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+            HStack {
+                Text(Constants.cvcTextName)
+                Spacer()
+            }
+            SecureField(Constants.cvcTextFieldPlaceholder,
+                        text: $settingsTabViewModel.cvcTextFieldText)
+                .onChange(of: settingsTabViewModel.cvcTextFieldText) { _ in
+                    settingsTabViewModel.setOnCardSecureCVC()
+                }
+
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift:207]:
```suggestion
```

File: RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+        }
+    }
+    
+    // MARK: - Private Methods
+    
+    private func makeDataPickerView(scrollViewProxy: ScrollViewProxy) -> some View {
+        VStack {
+            HStack(spacing: 140) {
+                HStack {
+                    Text("Month")
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift:218]:
```suggestion
                    Text()
```

File: RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+                        }
+                    } label: {
+                        Image(systemName: "chevron.right")
+                    }
+                    .frame(width: 10,
+                           height: 10)
+                    .rotationEffect(Angle.degrees(settingsTabViewModel.isMonthPickerShown ? 90 : 0))
+                }
+                HStack {
+                    Text("Year")
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift:233]:
```suggestion
                    Text()
```

File: RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+                    Button {
+                        withAnimation {
+                            settingsTabViewModel.isYearPickerShown.toggle()
+                            scrollViewProxy.scrollTo(
+                                settingsTabViewModel.isYearPickerShown ? 1 : 0, anchor: .center)
+                        }
+                    } label: {
+                        Image(systemName: "chevron.right")
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift:241]:
```suggestion
                        Image(systemName: )
```

File: RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+                    }
+                    .frame(width: 10,
+                           height: 10)
+                    .rotationEffect(Angle.degrees(settingsTabViewModel.isYearPickerShown ? 90 : 0))
+                }
+            }
+            .font(.title2)
+            
+            HStack {
+
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift:252]:
```suggestion
```

File: RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+                    .frame(width: 10,
+                           height: 10)
+                    .rotationEffect(Angle.degrees(settingsTabViewModel.isYearPickerShown ? 90 : 0))
+                }
+            }
+            .font(.title2)
+            
+            HStack {
+                
+                Picker("Month",
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift:253]:
```suggestion
                Picker(,
```

File: RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+                .opacity(settingsTabViewModel.isMonthPickerShown ? 1 : 0)
+                .onChange(of: settingsTabViewModel.expireMonth) { newValue in
+                    withAnimation {
+                        settingsTabViewModel.expireMonthOnCardName = newValue
+                        settingsTabViewModel.isMonthPickerShown = false
+                        scrollViewProxy.scrollTo(0, anchor: .center)
+                    }
+                }
+                
+                Picker("Year",
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift:270]:
```suggestion
                Picker(,
```

File: RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+        .alert(isPresented: $settingsTabViewModel.isAlertAddingCardShown) {
+            if settingsTabViewModel.cvcTextFieldText.count == Constants.cvcLength {
+                return Alert(title: Text(Constants.cardAddedAlertTitleName))
+            } else {
+                return Alert(title: Text(Constants.wrongCVCAlertName))
+            }
+        }
+    }
+}
+
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/StoreTabBar/Tabs/SettingsTabView.swift:315]: ///

File: RM_SUI/RM_SUI/StoreTabBar/ViewModel/SettingsTabViewModel.swift [hunk 9]
```diff
@@ -0,0 +1,94 @@
+//
+//  SettingsTabViewModel.swift
+//  RM_SUI
+//
+//  Created by Дмитрий Супрун on 28.01.23.
+//
+
+import SwiftUI
+
+final class SettingsTabViewModel: ObservableObject {
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/StoreTabBar/ViewModel/SettingsTabViewModel.swift:10]: ///

File: RM_SUI/RM_SUI/View/GreetingView.swift [hunk 10]
```diff
@@ -69,25 +94,48 @@ struct GreetingView: View {
...
+                        height: Constants.logoImageFrameWidthHeight
+                    )
+            case .empty:
+                ProgressView()
+                    .frame(
+                        width: Constants.logoImageFrameWidthHeight,
+                        height: Constants.logoImageFrameWidthHeight
+                    )
+            @unknown default:
+                            EmptyView()
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/View/GreetingView.swift:128]:
```suggestion
   EmptyView()
```

File: RM_SUI/RM_SUI/View/ProductInfoView.swift [hunk 11]
```diff
@@ -61,11 +63,32 @@ struct ProductInfoView: View {
...
+        
     }
-
+    
     // MARK: - Private Properties
-
+    
     @StateObject private var productInfoViewModel = ProductInfoViewModel()
+    
+    var magnificationGesture: some Gesture {
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/View/ProductInfoView.swift:74]:
```suggestion
private var magnificationGesture: some Gesture {
```

File: RM_SUI/RM_SUI/View/ProductInfoView.swift [hunk 12]
```diff
@@ -61,11 +63,32 @@ struct ProductInfoView: View {
...
+        MagnificationGesture()
+            .onChanged { value in
+                productInfoViewModel.magnificationScale = value
+            }
+            .onEnded { _ in
+                productInfoViewModel.magnificationScale = 1
+            }
+    }
+    
+    var tapHeartGesture: some Gesture {
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/View/ProductInfoView.swift:84]:
```suggestion
   private var tapHeartGesture: some Gesture {
```

File: RM_SUI/RM_SUI/View/SignInView.swift [hunk 13]
```diff
@@ -47,6 +48,9 @@ struct SignInView: View {
                     signUpLogInButtonView
                     forgotPasswordButtonView
                     checkVerification
+                    NavigationLink("", destination:
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/View/SignInView.swift:51]:
```suggestion
                    NavigationLink(, destination:
```

File: RM_SUI/RM_SUI/View/SignInView.swift [hunk 14]
```diff
@@ -130,15 +134,22 @@ struct SignInView: View {
     }
 
     private var passwordSecureTextFieldView: some View {
-        SecureField("", text: $signInViewModel.password)
-            .padding(.horizontal, Constants.horizontalPadding)
-            .font(.title2)
+            SecureField("", text: $signInViewModel.password)
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/View/SignInView.swift:137]:
```suggestion
            SecureField(, text: $signInViewModel.password)
```

File: RM_SUI/RM_SUI/ViewModel/SignInViewModel.swift [hunk 15]
```diff
@@ -22,6 +22,8 @@ final class SignInViewModel: ObservableObject {
     @Published var isSupportAlertShown = false
     @Published var isPhoneTextFieldFocused = false
     @Published var isPasswordTextFieldFocused = false
+    @Published var isShaking: CGFloat = 0
```
Comment (Vlaadkaaaa) [inline RM_SUI/RM_SUI/ViewModel/SignInViewModel.swift:25]:
```suggestion
    @Published var Shaking: CGFloat = 0
```
это же не флаг

**Review by AlexanderKokh — APPROVED** [review]

**Pull request merged**